Caused by: java.io.EOFException

For some reason, this is happening on CI builds but not locally.
## Command:
`./gradlew clean assembleDebug testDebug`

## Error:
```
:assembleDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':assembleDebug'.
> java.io.EOFException (no error message)

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':assembleDebug'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:203)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:185)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:66)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:50)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:25)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:110)
	at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
	at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
	at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
	at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
	at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:154)
	at org.gradle.internal.Factories$1.create(Factories.java:22)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:52)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:151)
	at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:32)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:99)
	at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:93)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:62)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:93)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:82)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter$DefaultBuildController.run(InProcessBuildActionExecuter.java:94)
	at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:43)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:28)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:75)
	at org.gradle.launcher.exec.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:45)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:52)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.DaemonHealthTracker.execute(DaemonHealthTracker.java:40)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:66)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.health.HintGCAfterBuild.execute(HintGCAfterBuild.java:41)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:246)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: java.io.EOFException
	at com.android.dexdeps.DexData.readBytes(DexData.java:479)
	at com.android.dexdeps.DexData.parseHeaderItem(DexData.java:86)
	at com.android.dexdeps.DexData.load(DexData.java:55)
	at com.android.dexdeps.DexData$load.call(Unknown Source)
	at com.getkeepsafe.dexcount.DexFile.<init>(DexFile.groovy:196)
	at com.getkeepsafe.dexcount.DexFile.<init>(DexFile.groovy)
	at com.getkeepsafe.dexcount.DexFile.extractDexFromAar(DexFile.groovy:115)
	at com.getkeepsafe.dexcount.DexFile.extractDexData(DexFile.groovy:58)
	at com.getkeepsafe.dexcount.DexFile$extractDexData.call(Unknown Source)
	at com.getkeepsafe.dexcount.DexMethodCountTask.generatePackageTree(DexMethodCountTask.groovy:234)
	at com.getkeepsafe.dexcount.DexMethodCountTask.countMethods(DexMethodCountTask.groovy:67)
	at com.getkeepsafe.dexcount.DexMethodCountTask$countMethods.call(Unknown Source)
	at com.getkeepsafe.dexcount.DexMethodCountPlugin$_applyAndroid_closure1_closure2_closure3.doCall(DexMethodCountPlugin.groovy:67)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:554)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:535)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
	... 68 more


BUILD FAILED

Total time: 27.37 secs
```

--- a/src/main/groovy/com/getkeepsafe/dexcount/DexCountException.groovy
+++ b/src/main/groovy/com/getkeepsafe/dexcount/DexCountException.groovy
@@ -1,0 +1,23 @@
+/*
+ * Copyright (C) 2016 KeepSafe Software
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.getkeepsafe.dexcount
+
+class DexCountException extends RuntimeException {
+    DexCountException() {}
+    DexCountException(String message) { super(message) }
+    DexCountException(String message, Throwable cause) { super(message, cause) }
+}

--- a/src/main/groovy/com/getkeepsafe/dexcount/DexFile.groovy
+++ b/src/main/groovy/com/getkeepsafe/dexcount/DexFile.groovy
@@ -17,9 +17,11 @@
 package com.getkeepsafe.dexcount
 
 import com.android.dexdeps.DexData
+import com.android.dexdeps.DexDataException
 import com.android.dexdeps.FieldRef
 import com.android.dexdeps.MethodRef
 
+import java.util.concurrent.TimeUnit
 import java.util.zip.ZipEntry
 import java.util.zip.ZipException
 import java.util.zip.ZipFile
@@ -52,12 +54,12 @@ class DexFile {
      * @return a list of DexFile objects representing data in the given file.
      */
     static List<DexFile> extractDexData(File file, int dxTimeoutSecs) {
-        try {
-            // AAR files need special treatment
-            if (file.name.endsWith(".aar")) {
-                return extractDexFromAar(file, dxTimeoutSecs);
-            }
+        // AAR files need special treatment
+        if (file.name.endsWith(".aar")) {
+            return extractDexFromAar(file, dxTimeoutSecs);
+        }
 
+        try {
             return extractDexFromZip(file)
         } catch (ZipException ignored) {
             // not a zip, no problem
@@ -103,14 +105,26 @@ class DexFile {
         // ~/android-sdk/build-tools/23.0.3/dx --dex --output=temp.dex classes.jar
         def tempDex = File.createTempFile("classes", ".dex")
         tempDex.deleteOnExit()
+
         def sout = new StringBuilder(), serr = new StringBuilder()
         def dxCmd = dxExe.absolutePath + " --dex --output=" + tempDex.absolutePath + " " + tempClasses.absolutePath
         def proc = dxCmd.execute()
         proc.consumeProcessOutput(sout, serr)
-        proc.waitForOrKill(dxTimeoutSecs * 1000)
-        if (!tempDex.exists()) {
-            throw new Exception("Error converting classes.jar into classes.dex: $serr")
+
+        if (!proc.waitFor(dxTimeoutSecs, TimeUnit.SECONDS)) {
+            proc.destroyForcibly()
+            throw new DexCountException("dx timed out after $dxTimeoutSecs seconds")
         }
+
+        def exitCode = proc.exitValue()
+        if (exitCode != 0) {
+            throw new DexCountException("dx exited with exit code $exitCode\nstderr=$serr")
+        }
+
+        if (!tempDex.exists()) {
+            throw new DexCountException("Error converting classes.jar into classes.dex: $serr")
+        }
+
         // return resulting dex file in a list
         return [ new DexFile(tempDex, true) ]
     }
@@ -193,7 +207,12 @@ class DexFile {
         this.isInstantRun = isInstantRun
         this.raf = new RandomAccessFile(file, 'r')
         this.data = new DexData(raf)
-        data.load()
+
+        try {
+            data.load()
+        } catch (EOFException | DexDataException e) {
+            throw new DexCountException("Error loading dex file", e)
+        }
     }
 
     def List<MethodRef> getMethodRefs() {

--- a/src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountTask.groovy
+++ b/src/main/groovy/com/getkeepsafe/dexcount/DexMethodCountTask.groovy
@@ -64,11 +64,18 @@ class DexMethodCountTask extends DefaultTask {
 
     @TaskAction
     void countMethods() {
-        generatePackageTree()
-        printSummary()
-        printFullTree()
-        printChart()
-        printTaskDiagnosticData()
+        try {
+            generatePackageTree()
+            printSummary()
+            printFullTree()
+            printChart()
+            printTaskDiagnosticData()
+        } catch (DexCountException e) {
+            withStyledOutput(StyledTextOutput.Style.Error, LogLevel.ERROR) { out ->
+                out.println("Error counting dex methods.  Please contact the developer at https://github.com/KeepSafe/dexcount-gradle-plugin/issues")
+                out.exception(e)
+            }
+        }
     }
 
     static def percentUsed(int count) {

